### PR TITLE
{base}[GCCcore/12.2.0] git-annex v10.20230802

### PR DESCRIPTION
**Pull request by jkuhl-uni**
Hi there, this is a module providing git-annex. This will be useful together with my other PRs 
- #18672 
- #18645

to build Datalad.

### CODE REVIEW

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+build_cmd += "make install-bins BUILDER=stack PREFIX=%(builddir)s"
+
+files_to_copy = [(['git-annex','git-annex-shell'], 'bin'),
+		
+		]
+
+sanity_check_paths = {
+    'files': ['bin/git-annex', 'bin/git-annex-shell'],
+	'dirs' : ['bin'],
+}
```
Comment (smoors) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb:39]:
```suggestion
sanity_check_paths = {
    'files': ['bin/git-annex', 'bin/git-annex-shell'],
    'dirs' : [],
}
```

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+sanity_check_commands = ['git-annex version']
```
Comment (smoors) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb:41]: please also add a `moduleclass` here?
```
moduleclass = 'tools'
```

**Review by smoors — COMMENTED** [review]
please fix formatting: don't use tabs, and 4 spaces per indent.

**Comment by jkuhl-uni** [comment]
Hi, I understand the error, I just want to raise the issue that easybuild 4.8.0 (at least on our cluster) does not seem to check this, which is why I was not aware of this problem.

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                   'tag': '%(version)s',
+                   'clone_into': '%(name)s-%(version)s',
+                   },
+    'filename': '%(name)s-%(version)s.tar.gz',
+}]
+
+checksums = ['5c4cd3c0a1d141a1b85936fee3fa52f7638774c2146b138aa9ae218d3da4b89d']
+
+prebuildopts = "stack setup && stack build && "
+buildopts = "make install-bins BUILDER=stack PREFIX=%(builddir)s"
```
Comment (smoors) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb:32]: `buildopts` comes after the `make` command, so you should not repeat `make` here
```suggestion
buildopts = "install-bins BUILDER=stack PREFIX=%(builddir)s"
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by jkuhl-uni** [comment]
Okay, checksum does not match... I'll have a look again. I think I know what is going on, I think I forgiot the checkout of the correct tag when computing the given checksum.
Sorry about that

**Comment by jkuhl-uni** [comment]
Thanks, yes. I did that.

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+sources = [{
+    'git_config': {'url': 'git://git-annex.branchable.com',
+                   'repo_name': '%(name)s',
+                   'tag': '%(version)s',
+                   'clone_into': '%(name)s-%(version)s',
+                   },
+    'filename': '%(name)s-%(version)s.tar.gz',
+}]
+
+checksums = ['62e4b4d9ef5ecf587a854a761927b1dbf4d646a0b8570e594ff48e3dac3719b5']
```
Comment (smoors) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb:29]: sorry, should have spotted this earlier, but with git clones you should not add a checksum, as those are not guaranteed to be the same
```suggestion
checksums = [None]
```

Comment (jkuhl-uni) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.2.0.eb:29]: Ah okay, I was wonndering a little, but didn't question it. I'll commit it as suggested.

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @jkuhl-uni!

**Pull request merged**